### ISSUE TITLE
the approval flow is not triggered when adding the pendding tag Via the Files Automated Tagging App

### ISSUE THREAD

**Issue by eco-villenet** [nextcloud/approval#14]
the approval flow is not triggered when adding the pendding tag Via the Files Automated Tagging App.
Files Automated Tagging App add the tag, the file is shown waiting for approval, but the file is not shared with the user enabled to approve.
![daniel-files](https://user-images.githubusercontent.com/89955563/131774023-bc18a719-ab53-4031-8726-3b806b55c993.png)
![approval-manual](https://user-images.githubusercontent.com/89955563/131774026-ea1d2e80-98ed-4d99-952d-97e152d7a813.png)
![approval](https://user-images.githubusercontent.com/89955563/131774028-bd63cdc7-9f2f-4f5f-97a2-4c62ecc08cf6.png)
![tag](https://user-images.githubusercontent.com/89955563/131774031-6f160bf7-0f61-4453-b25f-350b72c59cef.png)
![flow](https://user-images.githubusercontent.com/89955563/131774034-84f6ee9c-e122-4a45-9ddb-3666f03fea87.png)

**Comment by eneiluj** [comment]
Thanks for the bug report.

It was quite tricky to fix that. I think it's fine now. Could you try latest nightly (v1.0.7-2-nightly) and confirm?

What's been done/fixed:
* when a request is done via tag assignment (manual or automated tagging), make sure everything is done like a "UI" request
* the requesting user is the file owner
* adapt notifications and activities to those changes

**Comment by eco-villenet** [comment]
@eneiluj Thank you very much for your prompt attention to this bug. I have tested the changes, it has improved a bit.
Now when assigning tag automatically, the flow seems to work well, file is shared, notes are added, and in UI everything looks good.
On the side of the user who must approve, a notification is received, the file is seen, but it is corrupt. I leave image.
![auto](https://user-images.githubusercontent.com/89955563/131916585-0fe9d3e8-d1f7-4d41-8759-751fc180043d.png)
![auto1](https://user-images.githubusercontent.com/89955563/131916601-a6725ef9-81f3-40b3-b179-48c5bb4b4294.png)

The request by UI works without problems.

**Comment by eneiluj** [comment]
I can't reproduce the problem. On the approver side, information is loaded successfully for the shared file in the sidebar.
Do you see related errors in the browser console log? Or in nextcloud.log?

**Comment by eco-villenet** [comment]
I do not know what is happening, in the logs I do not see anything strange when uploading a file and triggering the auto tag. I only see that in the file detail there is already something strange (
You requested approval on test2-APPROVE.pdf.part)
![detail](https://user-images.githubusercontent.com/89955563/131950392-7f3d2ca1-e4b8-4937-acd0-9a3e4c350a67.png)


Attached nextcloud log of a file upload process and automatic tag assignment and approval request.
[nextcloud.log](https://github.com/nextcloud/approval/files/7103312/nextcloud.log)
[request-approval.log](https://github.com/nextcloud/approval/files/7103313/request-approval.log)


I only see the error in the nextcloud log when I try to see the detail from the approver side.

Will I have installed the version v1.0.7-2 wrong?

**Comment by eneiluj** [comment]
Oh, in your case, it seems the intermediate files created when uploading are automatically tagged so they trigger an approval request... It does not happen to me, with small or big files.

How are you uploading the files? Via the web interface? Via the Desktop client?

Could you create a file (like a new text document) including APROBAR in its name and check if it works fine?

**Comment by eco-villenet** [comment]
yes, all the tests were uploading a file via web interface.
Test by creating the document and assigning the name APPROVE at the time of creation. the result is the same.
![text](https://user-images.githubusercontent.com/89955563/132018946-9d37c7af-4cdc-4121-a3ef-117aa52e6bbf.png)


If I create the document with another name and then rename it, it works perfect.
![rename-file](https://user-images.githubusercontent.com/89955563/132018975-618102eb-c363-418e-ae18-d662e5b54d3d.png)
![approver](https://user-images.githubusercontent.com/89955563/132019000-da1a8a5c-9698-4d62-a78d-20bedc65e99c.png)


I am going to create a new instance of nextcloud and install directly to version 1.0.7-2. I tell you the results.

**Comment by eneiluj** [comment]
Thanks for your patience and precious feedback!

**Comment by eco-villenet** [comment]
sorry, try a fresh install of nextcloud, version 22.1.1
approval 1.0.7

Same result.
![new-nextcloud](https://user-images.githubusercontent.com/89955563/132036363-629bb146-732b-4afa-82ec-439bda7c0fd9.png)

**Comment by eco-villenet** [comment]
Hi, I've been doing more testing, not getting it to work. There is something that catches my attention, in the web UI, I can see that the name of the file is already corrupted (test2-APPROVE.pdf.part), but in the logs, there is no reference to that name, and that considering that I am in debug mode.

Another piece of information that I forgot to mention, in my configuration I am using the Oracle Bucket platform as storage. I don't know if that affects anything.

**Comment by eneiluj** [comment]
> Another piece of information that I forgot to mention, in my configuration I am using the Oracle Bucket platform as storage. I don't know if that affects anything.

Yeah it is most likely the reason why I can't reproduce the issue on my side. Could you setup a standard classic NC test instance?

If using an Oracle Bucket platform as storage is the cause of this remaining problem, then it has to be fixed on this side. There's not much I can do in the approval app which just handles the tag assignment event. This event should not be triggered for temporary partial files...
